### DISCUSSION TITLE
Add support for hx-target-* attribute (VSCode plugin)

### DISCUSSION THREAD

**Discussion by koddr** [a-h/templ#269] [Ideas]
Hi,

First, let me thank you for this amazing tool that allows you to develop the Go web applications very quickly! Love it 😘

I noticed that the [plugin for VSCode](https://github.com/a-h/templ-vscode) does not _full_ support the **htmx** extension [`response-targets`](https://htmx.org/extensions/response-targets/):

<img width="1240" alt="screenshot" src="https://github.com/a-h/templ/assets/11155743/ceb48342-b69c-4a80-ae35-2292ce469a10">

It would be really cool if you could add wildcard support. 

Right now, you have to define each `hx-target` like this:

```html
<form
 	hx-post="/api/secret/unlock"
 	hx-target="#secret"
 	hx-target-400="#errors"
 	hx-target-403="#errors"
 	hx-target-404="#errors"
 	hx-target-500="#errors"
 	hx-target-502="#errors"
 	hx-target-503="#errors"
 	hx-indicator="#loading-indicator"
>
...
```

Although it could be simplified to:

```html
<form
 	hx-post="/api/secret/unlock"
 	hx-target="#secret"
 	hx-target-*="#errors"
 	hx-indicator="#loading-indicator"
>
...
```

Thanks for your time and `templ` library! 😊

**Comment by a-h** [comment]
Thanks for your comments. I think we can turn this into an issue and work on it as a feature request.

Basically, the task is to update the attribute name parser to support having a `*` character in attribute names as per https://htmx.org/extensions/response-targets/#wildcard-resolution

It is a very straightforward change if you have time to do a PR yourself, if not, I'll get to it. 😁 

The attribute name parser is here:

https://github.com/a-h/templ/blob/e8f18b2fae1b2160869c1191ec3889332514a8e1/parser/v1/elementparser.go#L62-L68

Then, need to add a test for the new behaviours here:

https://github.com/a-h/templ/blob/e8f18b2fae1b2160869c1191ec3889332514a8e1/parser/v1/elementparser_test.go#L12

And finally, a test to make sure that the generated output is as expected by updating the `template.templ` file in https://github.com/a-h/templ/tree/main/generator/test-element-attributes, re-generating templates, and updating the `expected.html` file to include the expected output.

**Comment by a-h** [comment]
Added to templ in https://github.com/a-h/templ/commit/43545f88a52c9259c2a498a0914cbc2307d37c8f
